datastore/google/cloud/datastore/transaction.py

```diff
@@ -17,6 +17,48 @@
 from google.cloud.datastore.batch import Batch
 
 
+class TransactionOptions(object):
```

Comment:
I thought about that, and then decided to follow protobuf.  This could also be better in the case of a change upstream to mirror what they do.  I'll do whatever you guys decide on though.

Reply:
I'm not sure what you mean when you say "follow protobuf". As I see it, there is a class with 30+ lines of code accomplishing what a keyword argument could. It doesn't seem like a win for usability.

As for future-proofing it ("could also be better in the case of a change upstream") that is definitely a possibility. If they added **many** arguments, then having an `option=` keyword (vs. a `read_only=`) keyword would be beneficial. But that's a big if and not one worth taking into this design decision IMO.

Reply:
But my thought was that if protobuf just wanted a boolean, it would have specified a boolean.  Doesn't matter though.  I don't mind changing it.